apps/server/src/modules/account/domain/services/account.service.ts

```diff
@@ -256,18 +259,25 @@ export class AccountService extends AbstractAccountService implements DeletionSe
 
 	async save(accountSave: AccountSave): Promise<Account> {
 		const ret = await this.accountDb.save(accountSave);
-		const newAccount: AccountSave = {
+		const newAccount = new AccountSave({
 			...accountSave,
 			id: accountSave.id,
 			idmReferenceId: ret.id,
 			password: accountSave.password,
-		};
+		});
 		const idmAccount = await this.executeIdmMethod(async () => {
 			this.logger.debug(new SavingAccountLoggable(ret.id));
 			const account = await this.accountIdm.save(newAccount);
 			this.logger.debug(new SavedAccountLoggable(ret.id));
 			return account;
 		});
+
+		if (this.configService.get('FEATURE_IDENTITY_MANAGEMENT_STORE_ENABLED') === true) {
+			if (idmAccount === null || idmAccount?.username !== accountSave.username) {
```

Comment:
test coverage cover one of the conditions here not both

Reply:
Done